jormungandr/src/fragment/process.rs

```diff
@@ -47,26 +48,46 @@ impl Process {
         }
     }
 
-    pub async fn start<P: Into<PathBuf>>(
+    pub async fn start<P: AsRef<Path>>(
         self,
         n_pools: usize,
         service_info: TokioServiceInfo,
         stats_counter: StatsCounter,
         mut input: MessageQueue<TransactionMsg>,
         persistent_log_dir: Option<P>,
     ) -> Result<(), Error> {
-        let persistent_log = match persistent_log_dir {
+        async fn hourly_wakeup(enabled: bool) {
+            if enabled {
+                let now = Utc::now();
+                let current_hour = now.duration_trunc(Duration::hours(1)).unwrap();
+                let next_hour = current_hour + Duration::hours(1);
+                let sleep_duration = (next_hour - now).to_std().unwrap();
+                tokio::time::sleep(sleep_duration).await
```

Comment:
Could this be rewritten as `tokio::time::interval_at(next_hour_rounded, Duration::from_secs(60 * 60))`

Reply:
I was a bit concerned about clock or wakeup drift, as well as potential issues with machine suspend and similar when I wrote this. It's better to calculate the precise duration towards the next hour every time.